zeebe/gateway-protocol/src/main/proto/rest-api.yaml

```diff
@@ -678,6 +713,121 @@ components:
         endDate:
           type: string
           format: date-time
+    UserTaskSearchQueryRequest:
+      allOf:
+        - $ref: "#/components/schemas/SearchQueryRequest"
+      description: User task search query request.
+      type: object
+      properties:
+        filter:
+          $ref: "#/components/schemas/UserTaskFilterRequest"
+    UserTaskSearchQueryResponse:
+      allOf:
+        - $ref: "#/components/schemas/SearchQueryResponse"
+      description: User task search query response.
+      type: object
+      properties:
+        items:
+          type: array
+          items:
+            $ref: "#/components/schemas/UserTaskItem"
+    UserTaskFilterRequest:
+      description: User task filter request.
+      type: object
+      properties:
+        key:
+          type: integer
+          format: int64
+        taskState:
+          type: string
+        assignee:
+          type: string
+        taskDefinitionId:
+          type: string
+        candidateGroup:
+          type: string
+        candidateUser:
+          type: string
+        processDefinitionKey:
+          type: integer
+          format: int64
+        processInstanceKey:
+          type: integer
+          format: int64
+        creationTime:
+          $ref: "#/components/schemas/DateFilter"
+        completionTime:
+          $ref: "#/components/schemas/DateFilter"
+        followUpDate:
+          $ref: "#/components/schemas/DateFilter"
+        dueDate:
+          $ref: "#/components/schemas/DateFilter"
+        variables:
+          type: array
+          items:
+            $ref: "#/components/schemas/VariableFilter"
+        tenantIds:
+          type: string
+    UserTaskItem:
```

Comment:
🔧 All the information [provided by Zeebe itself at runtime](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.p886hph1hpur) should be reflected here, too. The task search result might contain more information on top (e.g. a `completionDate`) but not less. Please make sure that all runtime information is also present in the search result.

Reply:
Note this is imported by tasklist-task indices, and on this version, on the tasklist-task we dont have the external form over there, this will be added in a follow-up PR

Reply:
OK, I see. We need to be careful with this. I would rather say let's create the right REST model already, containing all information that is available from the runtime data. If we cannot fill this yet in the code then, let's add TODOs there so we don't forget to add it if we really cannot get this data yet from the ES/OS data model.